packages/ra-core/src/util/Query.tsx

```diff
@@ -0,0 +1,80 @@
+import { Component, ReactNode } from 'react';
+import withDataProvider from './withDataProvider';
+
+type DataProviderCallback = (
+    type: string,
```

Comment:
Maybe we can explicitly allow only the query types here

Reply:
Same answer as above